src/m3ninx/index/segment/builder/builder.go

```diff
@@ -518,14 +518,17 @@ func (b *builder) TermsIterable() segment.TermsIterable {
 }
 
 func (b *builder) FieldsPostingsList() (segment.FieldsPostingsListIterator, error) {
-	b.status.RLock()
-	defer b.status.RUnlock()
+	// NB(r): Need write lock since sort in newOrderedFieldsPostingsListIter
```

Comment:
nit: maybe `newOrderedFieldsPostingsListIter` -> `newOrderedFieldsPostingsListIterWithLock` to signify this too?